### PR TITLE
feat(doc): add pull request template

### PR DESCRIPTION
**Pull request by neubs-bsi**
Added a pull request template markdown file to root.

### CODE REVIEW

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# PULL REQUEST TEMPLATE
```
Comment (blaumeiser-at-bosch) [inline PULL_REQUEST_TEMPLATE.md]: You know, that you can add also PR templates under .github. Just search in Google for github issue and pr templates to learn how. The issue templates you can already find in this folder in the repo!

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# PULL REQUEST TEMPLATE
+
+## Description
+> Please provide a summary of your changes here. 
+> * Which issue is this pull request belonging to? (*Refer to issue here*)
```
Comment (sschuberth) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: We should probably point out to follow the syntax for [auto-closing keywords](https://help.github.com/en/articles/closing-issues-using-keywords) here.

As a site note, I personally prefer to have issue references in the commit messages, not in the PR description / discussion.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+## How Has This Been Tested?
+- [ ] Specific Unit Test
+- [ ] Executing the Example Projects
+- [ ] `mvn test` of whole project
+- [ ] `mvn test` of submodule
+- [ ] `gradle test` (for the gradle plugin)
+
+## Checklist
+- [ ] My code follows the style and guidlines of this project
```
Comment (sschuberth) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: "guidelines"

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+- [ ] I have self-reviewed my code 
+- [ ] I have provided tests that prove my change is working 
+- [ ] Existing tests still pass with my changes 
+- [ ] I have updated the documentation accordingly to my changes
+- [ ] My changes generate no new warnings
```
Comment (sschuberth) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: IMO this is something that should be technically enforced.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -144,7 +144,7 @@ The process described here has several goals:
 - Maintain Antenna's quality
 - Enable a sustainable system for maintainers to review contributions
 
-Please follow these steps to have your contribution considered by the maintainers:
+Please follow these steps, detailed in the [pull request template](.github/PULL_REQUEST_TEMPLATE/pull_request_template.md) to have your contribution considered by the maintainers:
```
Comment (sschuberth) [inline CONTRIBUTING.md]: Nit: Comma after closing parenthesis.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 4 above)
Comment (neubs-bsi) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: How would that be done? As a check run when doing a PR? Or is that also something github actions does?
And how would you technically enforce that the documentation is changed or that test have been provided? 
The tests that are still passing is obviously controlled. And checking for warnings is also doable.

Comment (sschuberth) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: I was only referring to the "My changes generate no new warnings" line, and I'd assume there's a way to make Maven fail a build (and thus CI) if a threshold of *X* warnings is exceeded (if you talk about compiler warnings; but similarly for warnings from static code analysis).

**Comment by neubs-bsi** [comment]
\cc @blaumeiser-at-bosch @sschuberth

**Pull request merged**